SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp

```diff
@@ -0,0 +1,484 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/RPCs/SpatialRPCService.h"
+
+#include "EngineClasses/SpatialNetBitReader.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "SpatialConstants.h"
+#include "Utils/RepLayoutUtils.h"
+#include "Utils/SpatialLatencyTracer.h"
+
+DEFINE_LOG_CATEGORY(LogSpatialRPCService);
+
+namespace SpatialGDK
+{
+SpatialRPCService::SpatialRPCService(const ExtractRPCDelegate InExtractRPCCallback, const FSubView& InActorAuthSubView,
+									 const FSubView& InActorNonAuthSubView, USpatialNetDriver* InNetDriver, FTimerManager* InTimerManager)
+	: ExtractRPCCallback(InExtractRPCCallback)
+	, NetDriver(InNetDriver)
+	, TimerManager(InTimerManager)
+	, RPCStore(FRPCStore())
+	, ClientServerRPCs(InExtractRPCCallback, InActorAuthSubView, InNetDriver, RPCStore)
+	, MulticastRPCs(InExtractRPCCallback, InActorNonAuthSubView, RPCStore)
+	, AuthSubView(&InActorAuthSubView)
+{
+	PeriodicallyProcessIncomingRPCs();
+	IncomingRPCs.BindProcessingFunction(FProcessRPCDelegate::CreateRaw(this, &SpatialRPCService::ApplyRPC));
+}
+
+void SpatialRPCService::Advance()
+{
+	ClientServerRPCs.Advance();
+	MulticastRPCs.Advance();
+	// TODO: Cross server should go here too
+}
+
+void SpatialRPCService::ProcessIncomingRPCs()
+{
+	IncomingRPCs.ProcessRPCs();
```

Comment:
could we simplify the RPC Container and just use it as a simple data store, while the processing logic will live here?

Reply:
Yes! Well, probably. I'll try it, and if it's nice I'll add it to this PR

Reply:
hmm ok so there are two instances used, one for incoming and one for outgoing. I think we should absolutely simplify this flow so the incoming and outgoing for RPCs is all handled together in one service, at which point making the container an actual container would be awesome. However at first glance it seems a little big for this PR

Reply:
Had similar thoughts. I think we can turn most of the rpc service into a series of queues + writers / readers but it seems like it will then require a load more tests and be a lot more work.